src/Symfony/Component/Workflow/Dumper/PlantUmlDumper.php

```diff
@@ -0,0 +1,171 @@
+    const TRANSITION_TYPES = array(self::DEFAULT_TRANSITION_TYPE, self::SQUARE_TRANSITION_TYPE);
+    const DEFAULT_OPTIONS = array(
+        'skinparams' => array(
+            'titleBorderRoundCorner' => 15,
+            'titleBorderThickness' => 2,
+            'state' => array(
+                'BackgroundColor<<'.self::INITIAL.'>>' => '#87b741',
+                'BackgroundColor<<'.self::MARKED.'>>' => '#3887C6',
+                'BorderColor' => '#3887C6',
+                'BorderColor<<'.self::MARKED.'>>' => 'Black',
+                'FontColor<<'.self::MARKED.'>>' => 'White',
+            ),
+            'agent' => array(
+                'BackgroundColor' => '#ffffff',
+                'BorderColor' => '#3887C6',
+            ),
+        ),
+    );
+
+    private $transitionType = self::DEFAULT_TRANSITION_TYPE;
+
+    public function __construct(string $transitionType = null)
+    {
+        if (!in_array($transitionType, self::TRANSITION_TYPES)) {
+            throw new InvalidArgumentException("Transition type '{$transitionType}' does not exist.");
+        }
+        $this->transitionType = $transitionType;
+    }
+
+    public function dump(Definition $definition, Marking $marking = null, array $options = array()): string
+    {
+        $options = array_replace_recursive(self::DEFAULT_OPTIONS, $options);
+        $code = $this->initialize($options);
+        foreach ($definition->getPlaces() as $place) {
+            $code[] =
+                "state {$place}".
+                ($definition->getInitialPlace() === $place ? ' <<'.self::INITIAL.'>>' : '').
+                ($marking && $marking->has($place) ? ' <<'.self::MARKED.'>>' : '');
+        }
+        if ($this->isSquareTransitionType()) {
+            foreach ($definition->getTransitions() as $transition) {
+                $code[] = "agent {$transition->getName()}";
+            }
+        }
+        foreach ($definition->getTransitions() as $transition) {
+            foreach ($transition->getFroms() as $from) {
+                foreach ($transition->getTos() as $to) {
+                    if ($this->isSquareTransitionType()) {
+                        $lines = array(
+                            "{$from} --> {$transition->getName()}",
+                            "{$transition->getName()} --> {$to}",
+                        );
+                        foreach ($lines as $line) {
+                            if (!in_array($line, $code)) {
+                                $code[] = $line;
+                            }
+                        }
+                    } else {
+                        $code[] = "{$from} --> {$to}: {$transition->getName()}";
+                    }
+                }
+            }
+        }
+
+        return $this->startPuml($options).$this->getLines($code).$this->endPuml($options);
+    }
+
+    private function isSquareTransitionType(): bool
+    {
+        return static::SQUARE_TRANSITION_TYPE === $this->transitionType;
```

Comment:
use `self::` to access constant. There is no reason to make them an extension point by allowing changing their values (and the code would not work anyway in such case, due to other places)